oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/ReactiveOidcIdTokenDecoderFactory.java

```diff
@@ -89,6 +90,7 @@ public final class ReactiveOidcIdTokenDecoderFactory implements ReactiveJwtDecod
 	private Function<ClientRegistration, Converter<Map<String, Object>, Map<String, Object>>> claimTypeConverterFactory = (
 			clientRegistration) -> DEFAULT_CLAIM_TYPE_CONVERTER;
 
+	private WebClient webClient = WebClient.create();
```

Comment:
To remain consistent with the multi-tenant nature of this class, I think this should be `Function<ClientRegistration, WebClient>` instead. Are you able to make that change/

Reply:
done